---
title: "PanoramiR Analysis Report"
author: "MiRXES Lab"
date: "`r Sys.Date()`"
output:
  bookdown::html_document2:
    theme: paper
    toc: true
    number_section: true
    fig_caption: true
---
```{r echo = FALSE, warning= FALSE, message = FALSE}
# Preparation
source('scripts/config.R')           # Problem: cannot change the work directory in config.R
source('scripts/step1_preprocess.R') # Merge result files & First round filtering
source('scripts/step1.1_samplesheet.R') # Read in Samplesheet
```


# Introduction
MicroRNAs are a group of small noncoding RNAs that serve as key signalling molecules that regulate gene expression at the post-transcriptional level. miRNA expression analysis is an invaluable tool for biomarker development, which can be translated into early diagnosis, risk stratification, and therapeutic management tools ^[Chakraborty C et al. Therapeutic miRNA and siRNA: Moving from Bench to Clinic as Next Generation Medicine. Mol Ther Nucleic Acids 2017;8:132–143.].\

Extracellular miRNAs have been widely reported as potential bomarkers in a variety of diseases, including cancer, cardiovascular disease and other chronic diseases^[Krol J et al. The widespread regulation of microRNA biogenesis, function and decay. Nat Rev Genet 2010;11(9):597–610.], where they function as signalling messengers to mediate cell-cell communication ^[Hayes J et al. MicroRNAs in cancer: biomarkers, functions and therapy. Trends Mol Med 2014;20(8):460–469.].\

MiRXES offers the broadest microRNA technology platform to translate microRNA research into clinical applications, from biomarker discovery and validation to product development. 

** TO DO ** Change the content to PanoramiR
**MiRXES' premium biomarker profiling** is a biomarker discovery platform which uses its patented technology based on the real-time reverse transcription PCR technique to reliably measure microRNAs with high sensitivity and then accurately determine miRNA expressions in specimens. This technical service report provides a run-through of the analysis workflow, from sample isolation to miRNA profiling and data analysis.\


# Experimental Approach and workflow
376 miRNAs are profiled using MiRXES's modified stem-looped reverse transcription- qualitative PCR (mSMRT-qPCR) platform ^[pattern].The mSMRT-qPCR method shows high selectivity for mature but not precursor forms of miRNAs and specific detection of highly similar sequences. Each step of the workflow, from sample handling to data analysis, is carefully designed to confer assay sensitivity, reduce variation and provide highest assay quality with minimal variation.\

** TODO **: Check the Experimental Approach and workflow

# Pre-processing of Data

## Spike-in Normalization
** TODO ** Description: Be attention to the isolation spike-in and RT spike-in
** TODO ** Add formular here
```{r echo = FALSE, warning = FALSE, message=FALSE}
source('scripts/step2_RSInorm.R') # Calculate SP factor -> Normalization -> Second round filtering
```

# Summary of Experiment
## Number of miRNA Detected in Samples
Figure below shows the number of miRNAs detected in each samples.
```{r echo = FALSE, warning = FALSE, message = FALSE, fig.align='center', fig.cap= "Number of miRNA Detected in Samples", out.width='80%'}
ggplotly(p.num.miRNA) # generated in step2_RSInorm.R
```

## miRNA Expression Levels in Each Samples {.tabset .tabset-pills}
Figure below shows the miRNA expression levels in each samples. Values have been spike-in normalized.

### `r if('Comparison 1' %in% comparisons) {title.compare[1]}` {.unnumbered}
```{r echo = FALSE, warning= FALSE, message=FALSE, fig.align='center', fig.cap= "miRNA Ct values distribution of samples (Comparison 1)", out.width= '80%'}
if ("Comparison 1" %in% comparisons) {
  p.violin.1 = fun.plot.violin('Comparison 1', col.compare.1)
  ggplotly(p.violin.1)
}
```

### `r if('Comparison 2' %in% comparisons) {title.compare[2]}` {.unnumbered}
```{r echo = FALSE, warning= FALSE, message=FALSE, fig.align='center', fig.cap= "miRNA Ct values distribution of samples (Comparison 2)", out.width= '80%'}
if ("Comparison 2" %in% comparisons) {
  p.violin.2 = fun.plot.violin('Comparison 2', col.compare.2)
  ggplotly(p.violin.2)
}
```

### `r if('Comparison 3' %in% comparisons) {title.compare[3]}` {.unnumbered}
```{r echo = FALSE, warning= FALSE, message=FALSE, fig.align='center', fig.cap= "miRNA Ct values distribution of samples (Comparison 3)", out.width= '80%'}
if ("Comparison 3" %in% comparisons) {
  p.violin.3 = fun.plot.violin('Comparison 3', col.compare.3)
  ggplotly(p.violin.3)
}
```

## Missing Value Imputation
The miRNAs with missing values in more than 10% (> 10%) of the total number of samples are not analysed. Else, the missing values are imputed by the lowest copy number across the samples.\
**TODO** The Figure could be removed because most of the miRNA could be analyzed later
```{r echo = FALSE, warning=FALSE, fig.align='center', fig.cap = 'Number of non-missing values of miRNA across samples', out.width = '80%'}
source('scripts/step3_imputation.R')
ggplotly(p.num.noNA) 
```

## Global Normalization
Global normalization ^[Mestdagh P, Van Vlierberghe P, De Weer A, Muth D, Westermann F, Speleman F, Vandesompele J. A novel and universal method for microRNA RT-qPCR data normalization. Genome Biol. 2009;10(6):R64. doi: 10.1186/gb-2009-10-6-r64. Epub 2009 Jun 16. PMID: 19531210; PMCID: PMC2718498.] is performed following spike-in normalization. Global normalization is performed on the miRNAs from all the samples by ensuring that the arithmetic means Ct values from all the samples are the same. A normalization constant is added to the Ct values for each miRNA in the sample so that the arithmetic mean Ct values of all the samples becomes identical.\

\begin{equation}
\tag{1}
Avg_{sample.i} = \frac{\sum_{i = 1}^{n_{miRNA}} log2(Copy\ Number)_{i}}{n_{miRNA}}
\end{equation}

\begin{equation}
\tag{2}
Avg_{all} = \frac{\sum_{i = 1}^{n_{sample}} Avg_{sample,i}}{n_{sample}}
\end{equation}

\begin{equation}
\tag{2}
Global.Factor_i = Avg_{all} - Avg_{sample,i}
\end{equation}

1. $Avg_{sample.i}$ is the average log2 copy number for sample $i$ ($n_{miRNA}$ is the number of miRNA in sample $i$)
2. $Avg_{all}$ is the average log2 copy number for all samples ($n_{sample}$ is the number of samples)
3. $Global.Factor_i$ is the correction factor of sample $i$ for global normalization.
```{r echo = FALSE, warning=FALSE, message=FALSE}
source('scripts/step4_GlobalNorm.R') # Global Normalization
```

# Overview Information
```{r echo = FALSE, warning=FALSE, message=FALSE}
source('scripts/step4.1_OverviewInfo.R')
```

**This Part Could be included in the standard report**

## Principal Component Analysis {.tabset .tabset-pills}
Principal component analysis is performed on the normalized miRNA expression data to determine if there are any strong underlying patterns. PCA analysis for different comparisons is plotted respectively.
]
### `r if('Comparison 1' %in% comparisons) {title.compare[1]}` {.unnumbered}

```{r echo = FALSE, fig.align='center', fig.cap = 'PCA analysis for Comparison 1', warnning = FALSE, message= FALSE, out.width='80%'}
if('Comparison 1' %in% comparisons) {
  p.pca.1 = fun.pca('Comparison 1', col.compare.1)
  ggplotly(p.pca.1)
}
```

### `r if('Comparison 2' %in% comparisons) {title.compare[2]}` {.unnumbered}

```{r echo = FALSE, fig.align='center', fig.cap = 'PCA analysis for Comparison 2', warnning = FALSE, message= FALSE, out.width='80%'}
if('Comparison 2' %in% comparisons) {
  p.pca.2 = fun.pca('Comparison 2', col.compare.2)
  ggplotly(p.pca.2)
}
```

### `r if('Comparison 3' %in% comparisons) {title.compare[3]}` {.unnumbered}

```{r echo = FALSE, fig.align='center', fig.cap = 'PCA analysis for Comparison 3', warnning = FALSE, message= FALSE, out.width='80%'}
if('Comparison 3' %in% comparisons) {
  p.pca.3 = fun.pca('Comparison 3', col.compare.3)
  ggplotly(p.pca.3)
}
```

## Hierachical Clustering Analysis {.tabset .tabset-pills}
Hierarchical clustering is performed on the normalized miRNA expression data to group clusters of similar miRNA expression together.

### miRNA {.unnumbered}
```{r echo = FALSE, warning=FALSE, message=FALSE, fig.align='center', fig.cap='Clustergram analysis', warning=FALSE, message=FALSE, out.width='80%'}
p.heatmap.all = fun.heatmap(df.tmp, "Sample-miRNA", df.anno.all, color.anno.all)
```

### Sample-sample distance {.unnumbered}
```{r echo = FALSE, warning= FALSE, message= FALSE, fig.align='center', fig.cap='Sample-to-sample distance', warning=FALSE, out.width='80%'}
p.heatmap.s2s = pheatmap(sample.dist.matrix,
                         clustering_distance_row = sample.dist,
                         clustering_distance_col = sample.dist,
                         col = colors.heatmap,
                         annotation_colors = color.anno.all,
                         annotation_col = df.anno.all,
                         annotation_names_col = FALSE,
                         main = "Euclidean distance of samples")
```

# Differential Expression Analysis {.tabset .tabset-pills}
The top 20 miRNAs having the smallest p value between the sample groups are listed in the table below. dCt of each miRNA is expressed as the mean Ct value of that in Group A minus the mean Ct value of that in Group B. Significance level is tested using Student's T-test and expressed in p-value.
```{r echo = FALSE, message=FALSE, warning=FALSE}
source('scripts/step5_DEanalysis.R')
```

### `r if('Comparison 1' %in% comparisons) {title.compare[1]}` {.unnumbered}
```{r echo = FALSE, warning=FALSE, message=FALSE}
if("Comparison 1" %in% comparisons) {
    ls.diff.1 = fun.DEanalysis('Comparison 1')
    
    kbl(ls.diff.1$res.order[1:20, ],
        caption = "20 miRNA with smallest p value in Comparison 1") %>%
        kable_classic(full_width = T, html_font = "Cambria") %>%
        scroll_box(width = '100%', height = '250px')
}
```

### `r if('Comparison 2' %in% comparisons) {title.compare[2]}` {.unnumbered}
```{r echo = FALSE, warning=FALSE, message=FALSE}
if("Comparison 2" %in% comparisons) {
    ls.diff.2 = fun.DEanalysis('Comparison 2')
    
    kbl(ls.diff.2$res.order[1:20, ],
        caption = "20 miRNA with smallest p value in Comparison 2") %>%
        kable_classic(full_width = T, html_font = "Cambria") %>%
        scroll_box(width = '100%', height = '250px')
}
```

### `r if('Comparison 3' %in% comparisons) {title.compare[3]}` {.unnumbered}
```{r echo = FALSE, warning=FALSE, message=FALSE}
if("Comparison 3" %in% comparisons) {
    ls.diff.3 = fun.DEanalysis('Comparison 3')
    
    kbl(ls.diff.3$res.order[1:20, ],
        caption = "20 miRNA with smallest p value in Comparison 3") %>%
        kable_classic(full_width = T, html_font = "Cambria") %>%
        scroll_box(width = '100%', height = '250px')
}
```

## Scatter Plot {.tabset .tabset-pills}
```{r echo = FALSE, message=FALSE, warning=FALSE, fig.cap="Scatter Plots "}

```

